apps/web/src/auth/getToken.ts

```diff
@@ -1,5 +1,13 @@
+import { IS_EE_AUTH_ENABLED } from '../config/index';
+import { getCookies } from '../utils/utils';
 import { LOCAL_STORAGE_AUTH_TOKEN_KEY } from './auth.const';
 
 export function getToken(): string {
-  return localStorage.getItem(LOCAL_STORAGE_AUTH_TOKEN_KEY) || '';
+  const token = IS_EE_AUTH_ENABLED ? getCookies('__session') : localStorage.getItem(LOCAL_STORAGE_AUTH_TOKEN_KEY);
```

Comment:
The JWT token from Clerk should always be fetched from `const { getToken} = useAuth()` or `window.Clerk.session.getToken()`.

Reply:
I agree but `getToken()` from `useAuth()` is async and we call `getToken()` synchronously inside the now legacy api.client inside `getHeader()` - we previously agreed to change this after the API client gets reworked